# Heart Disease Dataset Analysis

In this dataset, we will **predict** whether the patient has a heart disease or not.

<!-- <p align="center">
<img src="penguin.png" width="480">
</p> -->

You can have a view at this site: [link](https://allisonhorst.github.io/palmerpenguins/reference/penguins_raw.html)

Source code I downloaded from: [Kaggle](https://www.kaggle.com/datasets/parulpandey/palmer-archipelago-antarctica-penguin-data?select=penguins_size.csv)

___

|Rating           |Best beginner dataset for null treating|
|-----------------|-|
|Difficulty       | <font size=5>  &#9733; &#9733; &#9734; &#9734; &#9734; </font> |
|Dataset size     | <font size=5>  &#9733; &#9734; &#9734; &#9734; &#9734; </font> |
|Interesting topic| <font size=5>  &#9733; &#9733; &#9733; &#9733; &#9733; </font> |
|Popularity       | <font size=5>  &#9733; &#9733; &#9733; &#9733; &#9733; </font> |

___

In [1]:
# Framework for data cleaning and data visualising
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Framework for machine learning
from imblearn.over_sampling import SMOTE

from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.metrics import confusion_matrix

from sklearn.linear_model import LogisticRegression

In [3]:
df = pd.read_csv("Heart.csv")
df

id  Age  Sex     ChestPain  RestBP  Chol  Fbs  RestECG  MaxHR  ExAng  \
0      1   63    1       typical     145   233    1        2    150      0   
1      2   67    1  asymptomatic     160   286    0        2    108      1   
2      3   67    1  asymptomatic     120   229    0        2    129      1   
3      4   37    1    nonanginal     130   250    0        0    187      0   
4      5   41    0    nontypical     130   204    0        2    172      0   
..   ...  ...  ...           ...     ...   ...  ...      ...    ...    ...   
298  299   45    1       typical     110   264    0        0    132      0   
299  300   68    1  asymptomatic     144   193    1        0    141      0   
300  301   57    1  asymptomatic     130   131    0        0    115      1   
301  302   57    0    nontypical     130   236    0        2    174      0   
302  303   38    1    nonanginal     138   175    0        0    173      0   

     Oldpeak  Slope   Ca        Thal  AHD  
0        2.3      3  0.0       fixed   No  
1        1.5      2  3.0      normal  Yes  
2        2.6      2  2.0  reversable  Yes  
3        3.5      3  0.0      normal   No  
4        1.4      1  0.0      normal   No  
..       ...    ...  ...         ...  ...  
298      1.2      2  0.0  reversable  Yes  
299      3.4      2  2.0  reversable  Yes  
300      1.2      2  1.0  reversable  Yes  
301      0.0      2  1.0      normal  Yes  
302      0.0      1  NaN      normal   No  

[303 rows x 15 columns]

___

## 1) A data's problem

*What did the drum say to the drumstick?*

***My heart beats for you.***

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   id         303 non-null    int64  
 1   Age        303 non-null    int64  
 2   Sex        303 non-null    int64  
 3   ChestPain  303 non-null    object 
 4   RestBP     303 non-null    int64  
 5   Chol       303 non-null    int64  
 6   Fbs        303 non-null    int64  
 7   RestECG    303 non-null    int64  
 8   MaxHR      303 non-null    int64  
 9   ExAng      303 non-null    int64  
 10  Oldpeak    303 non-null    float64
 11  Slope      303 non-null    int64  
 12  Ca         299 non-null    float64
 13  Thal       301 non-null    object 
 14  AHD        303 non-null    object 
dtypes: float64(2), int64(10), object(3)
memory usage: 35.6+ KB


___

This dataset contains null. However, we will observe each feature first.

0) `id`: The index of the patient. Pretty useless for modeling unless you want exact prediction accuracy.
1) `Age`: The age of the patients
2) `Sse`: The sex of the patients (1 for male, 0 for female)
3) `ChestPain`: The type of 
chest pa (typical, asymptomatic, nonanginal, nontypical)
4) `RestBP`: The )
resting blood press. It seems this feature is about systolic blood pressure (mm/Hg)
5) `Chol`: The re
secholesteroll( in m)
6) `Fbs`: If the /dl
fasting blood sugar > 120 , the value is 1, otherwise 0
7) `RestECG`: The g/dl
resting electrocardiographic resal ue s 
8) `Maximum`: The ,1,2)
maximum hea
9) `ExAng`: If hieved
ingexerciss/causes induce, the value is 1, otherwise 0
10) `Oldpeak`: ldpeak = ST depression induced by exercise relativ
11) ` to rest
the slope of the peak exercise 
12) `T segment
number of major vessels (0-3) colored by
13) `flourosopy
thal: 0 = normal; 1 = fixed defect; 2 = rever
14) sable defect

___